tools/RepoTasks/RepoTasks.RemoteWorker/AppDomainWorker.cs

```diff
@@ -161,61 +196,164 @@ internal Configuration BuildXmlConfiguration(IEnumerable<Type> cmdletTypList)
-                    : null;
+                if (_onlyMarkedProperties)
+                {
+                    var ps1XmlAttributes = Attribute.GetCustomAttributes(memberInfo, typeof(Ps1XmlAttribute));
+                    foreach (var attribute in ps1XmlAttributes)
+                    {
+                        var ps1XmlAttribute = (Ps1XmlAttribute) attribute;
+                        if (ps1XmlAttribute.GroupByThis)
+                        {
+                            if (groupByInfo == null)
+                            {
+                                groupByInfo = new GroupByInfo
+                                {
+                                    Label = ps1XmlAttribute.Label,
+                                    ScriptBlock = ps1XmlAttribute.ScriptBlock,
+                                    PropertyName = memberInfo.Name,
+                                    TargetView = ps1XmlAttribute.Target,
+                                };
+
+                                continue;
+                            }
+                            else
+                            {
+                                // duplicate!
+                                // FIXME: replace or ignore or throw?
+                            }
+                        }
+
+                        var position = ps1XmlAttribute.Position;
+
+                        if ((ps1XmlAttribute.Target & ViewControl.Table) != ViewControl.None)
+                        {
+                            uint? width = ps1XmlAttribute.TableColumnWidth;
+                            var tableColumnHeader = new TableColumnHeader
+                            {
+                                Label = ps1XmlAttribute.Label ?? memberInfo.Name,
+                                Width = width > 0 ? width : null,
+                            };
+
+                            var tableColumnItem = ps1XmlAttribute.ScriptBlock != null
+                                ? new TableColumnItem
+                                {
+                                    Alignment = Alignment.Left,
+                                    ScriptBlock = ps1XmlAttribute.ScriptBlock,
+                                }
+                                : new TableColumnItem
+                                {
+                                    Alignment = Alignment.Left,
+                                    PropertyName = memberInfo.Name,
+                                };
+
+                            if (position == Ps1XmlConstants.DefaultPosition)
+                            {
+                                tableHeaders.Add(tableColumnHeader);
+                                tableColumnItems.Add(tableColumnItem);
+                            }
+                            else
+                            {
+                                columnHeadersWithPosition.Add(Tuple.Create(tableColumnHeader, position));
```

Comment:
what is multiple entries come up with the same position? Should we throw an exception saying that multiple entries found with the same position?

Also, what if the user starts the position at a non zero number? or does not specify position on all attributes?